library/src/main/kotlin/build/buf/connect/protocols/GRPCInterceptor.kt

```diff
@@ -188,11 +188,10 @@ internal class GRPCInterceptor(
 
     private fun Headers.withGRPCRequestHeaders(): Headers {
         val headers = toMutableMap()
-        // Note that we do not comply with the recommended structure for user-agent:
-        // https://github.com/grpc/grpc/blob/v1.51.1/doc/PROTOCOL-HTTP2.md#user-agents
-        // But this behavior matches connect-web:
-        // https://github.com/bufbuild/connect-web/blob/v0.4.0/packages/connect-core/src/grpc-web-create-request-header.ts#L33-L36
-        headers[GRPC_USER_AGENT] = listOf("@bufbuild/connect-kotlin")
+        if (headers.keys.none { it.equals(USER_AGENT, ignoreCase = true) }) {
+            // https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#user-agents
+            headers[USER_AGENT] = listOf("grpc-kotlin-connect/${ConnectConstants.VERSION}")
```

Comment:
no strong feeling, but I don't think connect-es/connect-go differentiate the user-agent based on protocol - seems helpful from a debug perspective, just thought I'd call it out as different.

(I think I'd expect maybe `connect-kotlin-grpc`, if we were going to include the protocol, but I guess `grpc-<xxx>` is closer to what the various grpc libraries add?)

Reply:
Made this change based on connect-go's implementation, alongside the two gRPC specs (which state that it should be `grpc-{language}-{variant}`). I don't feel strongly about this though.

Reply:
ah, yeah, I must have missed it looking through connect-go's source. thanks!